### PR TITLE
Integration spec

### PR DESCRIPTION
**Pull request by chepkok3**
In this PR:

- Integration tests with Capybara gem were added.
-  The app was secured from n+1 problem.

- [ ] Screenshot before and after fix N+1 problem
- Before
Before
- After 
After

### CODE REVIEW

File: spec/features/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
+require 'rails_helper'
+RSpec.describe 'User post index page:', type: :feature do
+  before(:each) do
```
Comment (Owusu-Desmond) [inline spec/features/post_index_spec.rb:3]: - [ ] Good that you have all your views test passing :+1: But please ensure all your application tests are passing to ensure your app works properly. :100: Currently you have some of your tests failing so please kindly fix them :pray: 
  
  image

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @chepkok3 and @hienphan0111 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 You have all your view tests passing :+1: ✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/post_index_spec.rb (hunk 1 above)
Comment (chepkok3) [reply spec/features/post_index_spec.rb:3]: @Owusu-Desmond Okay, let me fix them.

Comment (chepkok3) [reply spec/features/post_index_spec.rb:3]: @Owusu-Desmond Done

**Review by brytebee — APPROVED** [review]
Hi, @chepkok3 👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 

##

###  **Optional Suggestions:**
- N/A

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by chepkok3** [comment]
@hienphan0111 Thanks

**Pull request merged**